shopfloor/tests/test_delivery_scan_deliver.py

```diff
@@ -19,17 +19,59 @@ def setUpClassBaseData(cls):
                 (cls.product_d, 10),
                 # E as raw product with a lot
                 (cls.product_e, 10),
+                # F in two different packages
+                (cls.product_f, 10),
             ]
         )
         cls.pack1_moves = picking.move_lines[:2]
         cls.pack2_move = picking.move_lines[2]
+        cls.pack3_move = picking.move_lines[5]
         cls.raw_move = picking.move_lines[3]
         cls.raw_lot_move = picking.move_lines[4]
         cls._fill_stock_for_moves(cls.pack1_moves, in_package=True)
         cls._fill_stock_for_moves(cls.pack2_move, in_package=True)
         cls._fill_stock_for_moves(cls.raw_move)
         cls._fill_stock_for_moves(cls.raw_lot_move, in_lot=True)
+        # Set a lot for A for the variegated package (A + B)
+        cls.product_a_lot = (
+            cls.env["stock.production.lot"]
+            .sudo()
+            .create({"product_id": cls.product_a.id, "company_id": cls.env.company.id})
+        )
+        cls.product_a_quant = cls.env["stock.quant"].search(
+            [("product_id", "=", cls.product_a.id)]
+        )
+        cls.product_a_quant.sudo().lot_id = cls.product_a_lot
+        # Fill stock for F moves (two packages)
+        for __ in range(2):
+            product_f_pkg = cls.env["stock.quant.package"].create({})
+            cls._update_qty_in_location(
+                cls.pack3_move.location_id,
+                cls.pack3_move.product_id,
+                5,
+                package=product_f_pkg,
+            )
         picking.action_assign()
+        # Some records not related at all to the processed picking
+        cls.free_package = (
+            cls.env["stock.quant.package"].sudo().create({"name": "FREE_PACKAGE"})
+        )
+        cls.free_lot = (
+            cls.env["stock.production.lot"]
+            .sudo()
+            .create(
+                {
+                    "name": "FREE_LOT",
+                    "product_id": cls.product_a.id,
+                    "company_id": cls.env.company.id,
+                }
+            )
+        )
+        cls.free_product = (
+            cls.env["product.product"]
+            .sudo()
```

Comment:
```suggestion
```

Reply:
This one still requires `sudo` sadly, it is also done this way in the upstream `setupClassBaseData` to create products.

Reply:
yes! I misread the model :/ (tired)